app/controllers/concerns/foreman/controller/authentication.rb

```diff
@@ -6,12 +6,13 @@ def available_sso
   end
 
   def authenticate
-    return true if (User.current && Rails.env.test? && api_request?) ||
-                   session[:user] && (User.current = User.unscoped.find_by(id: session[:user]))
+    return true if User.current && Rails.env.test? && api_request?
+    User.current = User.unscoped.find_by(id: session[:user]) if session[:user]
+    return !User.current.disabled? if User.current
```

Comment:
we need to return only if we have the session[:user] set... if the User.current was set before, we don't care about such user - make sense as it is probably leftover from another request.
```suggestion
    return !User.current.disabled? if (session[:user] && User.current = User.unscoped.find_by(id: session[:user]))
```